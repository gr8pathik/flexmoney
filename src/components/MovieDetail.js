import axios from 'axios'
import React, {useEffect, useState} from 'react'
import { API_KEY, API_URL } from '../utils/Constants';

function MovieDetail({movieData}) {
    const [movieMetaData, setMovieMetaData] = useState("")
    useEffect(() => {
        async function fetchData() {
            const {data} = await axios.get(`${API_URL}?i=${movieData.imdbID}&apikey=${API_KEY}`)
            setMovieMetaData(data);
        }
        fetchData();
    }, [movieData.imdbID])

    return (<div className="movie-card" data-testid="movie-card">
        {movieData.Poster !== 'N/A' && <div className="movie-right" data-testid="movie-right">
            <img src={movieData.Poster} alt={movieData.Title} />
        </div>}
        <div className="movie-left">
            <div className="movie-header">
                <div className="movie-title" data-testid="movie-title">{movieData.Title}</div>
                <div className="movie-rating" data-testid="movie-rating"><span>&#9733;</span> {movieMetaData.imdbRating} / 10</div>
            </div>
            <div className="movie-info">{movieMetaData.Released} &#8226; {movieMetaData.Runtime} &#8226; {movieMetaData.Genre} &#8226; {movieMetaData.Language}</div>
            <div className="movie-meta"><b>Director:</b> {movieMetaData.Director}</div>
            <div className="movie-meta"><b>Writer:</b> {movieMetaData.Writer}</div>
            <div className="movie-meta"><b>Actors:</b> {movieMetaData.Actors}</div>
            <div className="movie-meta"><b>Plot:</b> {movieMetaData.Plot}</div>
        </div>
    </div>)
}

export default MovieDetail;