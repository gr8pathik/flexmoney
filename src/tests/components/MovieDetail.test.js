import { fireEvent, render, waitFor } from '@testing-library/react';
import axios from "axios";
import renderer from 'react-test-renderer';
import MovieDetail from '../../components/MovieDetail'

jest.mock("axios")

describe("SearchBox tests", () => {
    it('should render the movie data', async () => {
        const movieData = {
            "Title": "Madrid, int.",
            "Year": "2020",
            "imdbID": "tt12280634",
            "Type": "movie",
            "Poster": "https://m.media-amazon.com/images/M/MV5BMzhmNzE5YjItNGZhYi00ZjAzLTkyNjctODQzOTkxOGEyNzA2XkEyXkFqcGdeQXVyMTM2Mzg4MA@@._V1_SX300.jpg"
        };
        axios.get.mockResolvedValueOnce({
            data: {
                "Title": "Madrid, int.",
                "Year": "2020",
                "Rated": "N/A",
                "Released": "08 May 2020",
                "Runtime": "75 min",
                "Genre": "N/A",
                "Director": "Juan Cavestany",
                "Writer": "N/A",
                "Actors": "Lorena Iglesias, Petra del Rey, Carlos Martínez",
                "Plot": "N/A",
                "Language": "Spanish",
                "Country": "Spain",
                "Awards": "N/A",
                "Poster": "https://m.media-amazon.com/images/M/MV5BMzhmNzE5YjItNGZhYi00ZjAzLTkyNjctODQzOTkxOGEyNzA2XkEyXkFqcGdeQXVyMTM2Mzg4MA@@._V1_SX300.jpg",
                "Ratings": [
                    {
                        "Source": "Internet Movie Database",
                        "Value": "7.1/10"
                    }
                ],
                "Metascore": "N/A",
                "imdbRating": "7.1",
                "imdbVotes": "26",
                "imdbID": "tt12280634",
                "Type": "movie",
                "DVD": "N/A",
                "BoxOffice": "N/A",
                "Production": "N/A",
                "Website": "N/A",
                "Response": "True"
            }
          });  
        const { container, getByTestId } = render(<MovieDetail movieData={movieData} />)
        await waitFor(() => {
            expect(getByTestId("movie-right")).toBeInTheDocument();
            expect(getByTestId("movie-title")).toHaveTextContent("Madrid, int.")
            expect(getByTestId("movie-rating")).toHaveTextContent("★ 7.1 / 10")
        })
        expect(container).toMatchSnapshot()
    })
    it('should render the movie data without poster', async () => {
        const movieData = {
            "Title": "Madrid, int.",
            "Year": "2020",
            "imdbID": "tt12280634",
            "Type": "movie",
            "Poster": "N/A"
        };
        axios.get.mockResolvedValueOnce({
            data: {
                "Title": "Madrid, int.",
                "Year": "2020",
                "Rated": "N/A",
                "Released": "08 May 2020",
                "Runtime": "75 min",
                "Genre": "N/A",
                "Director": "Juan Cavestany",
                "Writer": "N/A",
                "Actors": "Lorena Iglesias, Petra del Rey, Carlos Martínez",
                "Plot": "N/A",
                "Language": "Spanish",
                "Country": "Spain",
                "Awards": "N/A",
                "Poster": "N/A",
                "Ratings": [
                    {
                        "Source": "Internet Movie Database",
                        "Value": "7.1/10"
                    }
                ],
                "Metascore": "N/A",
                "imdbRating": "7.1",
                "imdbVotes": "26",
                "imdbID": "tt12280634",
                "Type": "movie",
                "DVD": "N/A",
                "BoxOffice": "N/A",
                "Production": "N/A",
                "Website": "N/A",
                "Response": "True"
            }
          });  
        const { container, getByTestId, queryByTestId } = render(<MovieDetail movieData={movieData} />)
        await waitFor(() => {
            expect(queryByTestId("movie-right")).not.toBeInTheDocument();
            expect(getByTestId("movie-title")).toHaveTextContent("Madrid, int.")
            expect(getByTestId("movie-rating")).toHaveTextContent("★ 7.1 / 10")
        })
        expect(container).toMatchSnapshot()
    })
})
