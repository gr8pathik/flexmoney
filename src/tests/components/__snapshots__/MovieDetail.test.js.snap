// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox tests should render the movie data 1`] = `
<div>
  <div
    class="movie-card"
    data-testid="movie-card"
  >
    <div
      class="movie-right"
      data-testid="movie-right"
    >
      <img
        alt="Madrid, int."
        src="https://m.media-amazon.com/images/M/MV5BMzhmNzE5YjItNGZhYi00ZjAzLTkyNjctODQzOTkxOGEyNzA2XkEyXkFqcGdeQXVyMTM2Mzg4MA@@._V1_SX300.jpg"
      />
    </div>
    <div
      class="movie-left"
    >
      <div
        class="movie-header"
      >
        <div
          class="movie-title"
          data-testid="movie-title"
        >
          Madrid, int.
        </div>
        <div
          class="movie-rating"
          data-testid="movie-rating"
        >
          <span>
            ★
          </span>
           
          7.1
           / 10
        </div>
      </div>
      <div
        class="movie-info"
      >
        08 May 2020
         • 
        75 min
         • 
        N/A
         • 
        Spanish
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Director:
        </b>
         
        Juan Cavestany
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Writer:
        </b>
         
        N/A
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Actors:
        </b>
         
        Lorena Iglesias, Petra del Rey, Carlos Martínez
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Plot:
        </b>
         
        N/A
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchBox tests should render the movie data without poster 1`] = `
<div>
  <div
    class="movie-card"
    data-testid="movie-card"
  >
    <div
      class="movie-left"
    >
      <div
        class="movie-header"
      >
        <div
          class="movie-title"
          data-testid="movie-title"
        >
          Madrid, int.
        </div>
        <div
          class="movie-rating"
          data-testid="movie-rating"
        >
          <span>
            ★
          </span>
           
          7.1
           / 10
        </div>
      </div>
      <div
        class="movie-info"
      >
        08 May 2020
         • 
        75 min
         • 
        N/A
         • 
        Spanish
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Director:
        </b>
         
        Juan Cavestany
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Writer:
        </b>
         
        N/A
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Actors:
        </b>
         
        Lorena Iglesias, Petra del Rey, Carlos Martínez
      </div>
      <div
        class="movie-meta"
      >
        <b>
          Plot:
        </b>
         
        N/A
      </div>
    </div>
  </div>
</div>
`;
